backend/session/store.go

```diff
@@ -4,79 +4,153 @@
 )
 
-const sessionDataKey = "session_data"
+// MySQLStore represents the session store.
+type MySQLStore struct {
+	*sql.DB
+	stopCleanup chan bool
+}
 
-type Store struct {
-	wrappedStore wrappedsessionstore.DeletableSessionStore
+type SessionRow struct {
+	Data string `json:"session_data"`
 }
 
-type StoreOptions struct {
-	SessionDuration  time.Duration
-	UseSecureCookies bool
-	Key              []byte
+func GetSession(sessionManager *scs.SessionManager, r *http.Request) *Session {
+	sessionData := sessionManager.Get(r.Context(), config.SessionStoreKey())
+
+	if session, ok := sessionData.(*Session); ok {
+		return session
+	}
+
+	return &Session{}
 }
 
-func NewStore(db *database.Connection, opts StoreOptions) (*Store, error) {
-	wrappedStore, err := wrappedsessionstore.NewMySQLStoreFromConnection(
-		db.DB,
-		"/",
-		int(opts.SessionDuration.Seconds()),
-		opts.Key,
-	)
-	if err != nil {
-		return nil, err
+// New returns a new MySQLStore instance, with a background cleanup goroutine
+// that runs every 5 minutes to remove expired session data.
+func New(db *sql.DB) *MySQLStore {
+	return NewWithCleanupInterval(db, 12*time.Hour)
+}
+
+// NewWithCleanupInterval returns a new MySQLStore instance. The cleanupInterval
+// parameter controls how frequently expired session data is removed by the
+// background cleanup goroutine. Setting it to 0 prevents the cleanup goroutine
+// from running (i.e. expired sessions will not be removed).
+func NewWithCleanupInterval(db *sql.DB, cleanupInterval time.Duration) *MySQLStore {
+	m := &MySQLStore{
+		DB: db,
+	}
+
+	if cleanupInterval > 0 {
+		go m.startCleanup(cleanupInterval)
 	}
-	wrappedStore.SetSessionToUserID(userIDFromSession)
-	wrappedStore.Options.Path = "/"
-	wrappedStore.Options.HttpOnly = true
-	wrappedStore.Options.Secure = opts.UseSecureCookies
 
-	return &Store{wrappedStore}, nil
+	return m
 }
 
-func (store *Store) Read(r *http.Request) *Session {
-	sess := store.readRaw(r)
-	sessionData, ok := sess.Values[sessionDataKey].(*Session)
-	if ok {
-		return sessionData
+// Find returns the data for a given session id from the MySQLStore instance.
+// If the session id is not found or is expired, the returned exists flag will
+// be set to false.
+
+func (m *MySQLStore) Find(id string) ([]byte, bool, error) {
+	sess := SessionRow{}
+	row := m.DB.QueryRow("SELECT session_data FROM sessions WHERE id = ? AND UTC_TIMESTAMP(6) < expires_at", id)
+	err := row.Scan(&sess.Data)
+	if err == sql.ErrNoRows {
+		return nil, false, nil
+	} else if err != nil {
+		return nil, false, err
 	}
-	return &Session{}
+	fmt.Printf("sess: %+v\n", sess)
+	fmt.Println("data: ", sess.Data)
+	byteData := []byte(sess.Data)
+	fmt.Println("byteData: ", byteData)
+
+	return byteData, true, nil
 }
 
-func (store *Store) Set(w http.ResponseWriter, r *http.Request, s *Session) error {
-	sess := store.readRaw(r)
-	sess.Values[sessionDataKey] = s
-	return sess.Save(r, w)
+// Same function SCS uses to decode session data
```

Comment:
didn't love that I had to copy a function from the scs library (since it could easily change), but due to their API I had to do this in order to save teh user ID into the database (since the user ID is how we are able to delete a session).